medusa/storage/google_storage.py

```diff
@@ -122,12 +122,16 @@ async def _upload_object(self, data: io.BytesIO, object_key: str, headers: t.Dic
                 self.config.bucket_name, object_key
             )
         )
+
+        storage_class = self.storage.get_storage_class(),
+        ex_header = {"storageClass": storage_class}
         resp = await self.gcs_storage.upload(
             bucket=self.bucket_name,
             object_name=object_key,
             file_data=data,
             force_resumable_upload=True,
             timeout=-1,
+            headers=ex_header,
```

Comment:
I could not get this to work with GCS. All the uploads I did ended up with the Standard storage class. Swithich the default bucket storage class mode to `Managed with Autoclass` did not help either.

It's like the GCS ignores the HTTP header. Even if I set it in the request, the response comes with ` 'storageClass': 'STANDARD',`